src/CTA.Rules.Metrics/MetricsTransformer.cs

```diff
@@ -26,6 +26,46 @@ internal static IEnumerable<GenericActionExecutionMetric> TransformGenericAction
             return genericActionExecutionsMetrics;
         }
 
+        internal static IEnumerable<TargetVersionMetric> TransformTargetVersions(MetricsContext context, ProjectResult projectResult)
+        {
+            var projectFile = projectResult.ProjectFile;
+            var targetVersions = projectResult.TargetVersions;
+            var targetVersionMetrics = new List<TargetVersionMetric>();
+            foreach (var targetVersion in targetVersions)
+            {
+                targetVersionMetrics.Add(new TargetVersionMetric(context, targetVersion, projectFile));
+            }
+
+            return targetVersionMetrics;
+        }
+
+        internal static IEnumerable<UpgradePackageMetric> TransformUpgradePackages(MetricsContext context, ProjectResult projectResult)
+        {
+            var projectFile = projectResult.ProjectFile;
+            var upgradePackages = projectResult.UpgradePackages;
+            var upgradePackagesMetric = new List<UpgradePackageMetric>();
+            foreach (var upgradePackage in upgradePackages)
+            {
+                upgradePackagesMetric.Add(new UpgradePackageMetric(context, upgradePackage, projectFile));
```

Comment:
is there a way we can also get source package version ? and in the metrics just add "a.b.c(v1.0)" replaced by "a.b.e(v.3.4)" something in this line ?

Reply:
Yes. already started to work on this with PA. I will create a new PR to address the new metrics.

Reply:
ok create new Issue and track and resolve the comment